state/protocol/badger/state.go

```diff
@@ -514,6 +514,33 @@ func (state *State) updateEpochMetrics(snap protocol.Snapshot) error {
 		return fmt.Errorf("could not update committed epoch final view")
 	}
 
+	currentEpochFinalView, err := snap.Epochs().Current().FinalView()
+	if err != nil {
+		return fmt.Errorf("could not update current epoch final view")
+	}
+	state.metrics.CurrentEpochFinalView(currentEpochFinalView)
+
+	// update dkg phase 1 final view
+	dkgPhase1FinalView, err := snap.Epochs().Current().DKGPhase1FinalView()
+	if err != nil {
+		return fmt.Errorf("could not update current dkg phase 1 final view")
+	}
+	state.metrics.CurrentDKGPhase1FinalView(dkgPhase1FinalView)
+
+	// update dkg phase 2 final view
+	dkgPhase2FinalView, err := snap.Epochs().Current().DKGPhase2FinalView()
+	if err != nil {
+		return fmt.Errorf("could not update current dkg phase 2 final view")
+	}
+	state.metrics.CurrentDKGPhase2FinalView(dkgPhase2FinalView)
+
+	// update dkg phase 3 final view
+	dkgPhase3FinalView, err := snap.Epochs().Current().DKGPhase3FinalView()
+	if err != nil {
+		return fmt.Errorf("could not update current dkg phase 3 final view")
+	}
+	state.metrics.CurrentDKGPhase3FinalView(dkgPhase3FinalView)
```

Comment:
```suggestion
	dkgPhase1FinalView, dkgPhase2FinalView, dkgPhase3FinalView, err := protocol.DKGPhaseViews(currEpoch)
	if err != nil {
		return fmt.Errorf("could not get dkg phase final view: %w", err)
	}

	state.metrics.CurrentDKGPhase1FinalView(dkgPhase1FinalView)
	state.metrics.CurrentDKGPhase2FinalView(dkgPhase2FinalView)
	state.metrics.CurrentDKGPhase3FinalView(dkgPhase3FinalView)
```

Can save a bit of space with https://github.com/onflow/flow-go/blob/3d3a19b423f9420ed3651501f2f803a116561306/state/protocol/convert.go#L150